packages/headless/project.json

```diff
@@ -1 +1,166 @@
+        "{projectRoot}/dist/definitions/**"
+      ],
+      "outputs": ["{projectRoot}/temp"],
+      "executor": "nx:run-commands",
+      "options": {
+        "command": "node ./scripts/extract-documentation.mjs",
+        "cwd": "packages/headless"
+      }
+    },
+    "cached:build:doc": {
+      "dependsOn": ["cached:build:esbuild", "cached:build:doc:extract"],
+      "inputs": ["{projectRoot}/doc-parser/**", "{projectRoot}/temp/**"],
+      "outputs": ["{projectRoot}/dist/parsed_doc.json"],
+      "executor": "nx:run-commands",
+      "options": {
+        "command": "ts-node --project ./doc-parser/tsconfig.build.json ./doc-parser/doc-parser.ts",
+        "cwd": "packages/headless"
+      }
+    },
+    "build": {
+      "dependsOn": ["cached:build:esbuild", "cached:build:doc"],
+      "inputs": [],
+      "outputs": [],
+      "executor": "nx:noop"
+    },
+    "dev:esbuild": {
+      "dependsOn": [],
+      "inputs": [],
+      "outputs": [],
+      "executor": "nx:run-commands",
+      "options": {
+        "command": "node esbuild.mjs dev",
+        "cwd": "packages/headless"
+      }
+    },
+    "dev:definitions": {
+      "dependsOn": [],
+      "inputs": [],
+      "outputs": [],
+      "executor": "nx:run-commands",
+      "options": {
+        "command": "tsc -p src/tsconfig.build.json -d --emitDeclarationOnly --declarationDir dist/definitions -w",
+        "cwd": "packages/headless"
+      }
+    },
+    "dev": {
+      "dependsOn": [],
+      "inputs": [],
+      "outputs": [],
+      "executor": "nx:run-commands",
+      "options": {
+        "commands": [
+          "nx run headless:\"dev:esbuild\"",
+          "nx run headless:\"dev:definitions\""
+        ],
+        "parallel": true
+      }
+    },
+    "cached:test:unit": {
+      "dependsOn": ["^build", "cached:build:doc"],
+      "inputs": [
+        "^source",
+        "source",
+        "{projectRoot}/dist/parsed_doc.json",
+        "tests"
+      ],
+      "outputs": [],
+      "executor": "@nrwl/jest:jest",
+      "options": {
+        "jestConfig": "packages/headless/jest.config.js",
+        "testPathIgnorePatterns": ["src/integration-tests"]
+      }
+    },
+    "test:integration": {
+      "dependsOn": ["^build", "cached:build:doc"],
+      "inputs": [],
+      "outputs": [],
+      "executor": "@nrwl/jest:jest",
+      "options": {
+        "jestConfig": "packages/headless/jest.config.js",
+        "testPathPattern": ["src/integration-tests"]
+      }
+    },
+    "test": {
+      "dependsOn": ["cached:test:unit", "test:integration"],
+      "inputs": [],
+      "outputs": [],
+      "executor": "nx:noop"
+    },
+    "test:watch": {
+      "dependsOn": [],
+      "inputs": [],
+      "outputs": [],
+      "executor": "@nrwl/jest:jest",
+      "options": {
+        "jestConfig": "packages/headless/jest.config.js",
+        "watch": true,
+        "color": true,
+        "silent": false
+      },
+      "configurations": {
```

Comment:
At some point, I had all `test` tasks combined into one, which is why there's these configurations here. I had to divide them since they're not all cached and they have different dependencies.

I kept the configurations here since they could be useful.

You can use a specific configuration like so:
```sh
~/repos/ui-kit $ npx nx run headless:"test:watch":unit
```
OR
```sh
~/repos/ui-kit $ npm run -w @coveo/headless test:watch -- --configuration=unit
```